hotline/species/tests/test_forms.py

```diff
@@ -0,0 +1,52 @@
+from django.test import TestCase
+from django.test.client import RequestFactory
+from model_mommy.mommy import make
+
+from hotline.users.models import User
+
+from ..forms import SpeciesSearchForm
+from ..models import Species
+
+
+class SpeciesFormsTest(TestCase):
+    """
+    Some unit tests to ensure forms in the species app work as they should.
+
+    """
+    def setUp(self):
+        super(SpeciesFormsTest, self).setUp()
+        make(Species, name="stuff")
+        self.user = User(first_name="foo", last_name="bar", email="foobar@example.com", is_staff=True)
+        self.user.set_password("foobar")
+        self.user.save()
+        self.client.login(username="foobar@example.com", password="foobar")
+
+    def test_get_queryset(self):
+        """
+        test to ensure initial queryset is the entire set of species model objects.
+        """
+        form = SpeciesSearchForm(user=self.user, data={})
+        queryset = form.get_queryset()
+        self.assertEqual(len(queryset), Species.objects.count())
+
+    def test_valid_form(self):
+        data = {
+            "querystring": "other",
+        }
+        form = SpeciesSearchForm(user=self.user, data=data)
+        self.assertTrue(form.is_valid())
+
+    def test_search(self):
+        # test object
+        other = make(Species, name="other")
+        # set it all up
+        request = RequestFactory().get("/", {'querystring': 'other'})
+        form = SpeciesSearchForm(request.GET, user=self.user)
+        # make sure the form is valid
+        self.assertTrue(form.is_valid())
+        results = form.search()
+        values = results.to_dict()
+        ids_list = values['query']['filtered']['filter']['ids']['values']
```

Comment:
I'm not entirely sure this is what you want to do in this test. `search()` will return an elasticsearch-dsl Search object, which you can call `.execute()` on to get results. You'd probably want to make sure the results contain the species you created.

Here's the confusing part, **because the default behavior of the test runner in this project is to fake any calls to elasticsearch**, you'll want your test class to subclass `elasticmodels.ESTestCase` and `TestCase`, so elasticsearch is actually used.

See https://github.com/PSU-OIT-ARC/elasticmodels#testing